data_pipeline/elife_article_xml/elife_article_xml_pipeline.py

```diff
@@ -0,0 +1,159 @@
+import base64
+from typing import Iterable, Any, Optional, Sequence, Tuple
+import logging
+import xml.etree.ElementTree as ET
+import requests
+
+
+from data_pipeline.elife_article_xml.elife_article_xml_config import (
+    ElifeArticleXmlConfig,
+    ElifeArticleXmlSourceConfig
+)
+from data_pipeline.utils.collections import iter_item_until_exception
+from data_pipeline.utils.data_store.bq_data_service import (
+    does_bigquery_table_exist,
+    get_single_column_value_list_from_bq_query,
+    load_given_json_list_data_from_tempdir_to_bq
+)
+from data_pipeline.utils.json import (
+    get_recursively_transformed_object,
+    remove_key_with_null_value
+)
+from data_pipeline.utils.xml import parse_xml_and_return_it_as_dict
+
+LOGGER = logging.getLogger(__name__)
+
+
+class GitHubRateLimitError(RuntimeError):
+    pass
+
+
+def get_json_response_from_url(url: str) -> Any:
+    response = requests.get(url=url)
+    if response.status_code == 403:
+        raise GitHubRateLimitError()
```

Comment:
maybe it is worth adding the response content as part of the message (then we would log it when we catch the exception)